src/z_generate_repo.prog.abap

```diff
@@ -306,139 +305,120 @@ CLASS lcl_generator IMPLEMENTATION.
+                         i_error_text        = |The schema for interface { intfname } could not be created. Error when transforming schema content from string to xstring| ##NO_TEXT
+                         ).
       ENDIF.
     ENDLOOP.
 
-    IF whole_aff_folder = abap_true.
-      "also serialize the two interfaces: if_aff_types_v1 and if_aff_oo_types_v1
-      DATA two_interfaces TYPE if_aff_object_file_handler=>tt_objects .
-      APPEND VALUE #( devclass = 'SAFF_CORE' obj_name = 'IF_AFF_TYPES_V1' obj_type = 'INTF' ) TO two_interfaces.
-      APPEND VALUE #( devclass = 'SEO_AFF' obj_name = 'IF_AFF_OO_TYPES_V1' obj_type = 'INTF' ) TO two_interfaces.
-
-      DATA replacing_names TYPE replacing_tab.
-      replacing_names = VALUE #(
-       ( to_be_replaced = `if_aff_types_v1`    replace_with = `zif_aff_types_v1` )
-       ( to_be_replaced = `if_aff_oo_types_v1` replace_with = `zif_aff_oo_types_v1` )
-      ) ##NO_TEXT ##NO_TEXT.
-
-      LOOP AT two_interfaces ASSIGNING FIELD-SYMBOL(<interf>).
-        DATA(intf_files2) = file_handler->serialize_objects( objects = VALUE #( ( <interf> ) ) log = aff_framework_log ).
-        add_aff_files_to_zip( files = intf_files2 filename = `` replacing_table_string = replacing_names ).
-      ENDLOOP.
+    IF p_readm = abap_true.
+      DATA(interfacename) = replace_names_in_string( content_as_string = to_lower( intfname ) replacing_table_string = replacing_table_string ).
+      IF object-example  IS NOT INITIAL.
+        DATA(examplename) = replace_names_in_string( content_as_string = to_lower( object-example ) replacing_table_string = replacing_table_string ).
+        DATA(readme) = VALUE rswsourcet(
+            ( |# { object-object_type } File Format| )
+            ( `` )
+            ( `File | Cardinality | Definition | Schema | Example` )
+            ( `:--- | :---  | :--- | :--- | :---` )
+            ( |`<name>.{ object_type_folder_name }.json` \| 1 \| [`{ interfacename }.intf.abap`](./type/{ interfacename }.intf.abap) \| [`{ to_lower( objecttype ) }-v{ format_version }.json`](./{ to_lower( objecttype ) }-v{ format_version }.json)| &&
+              | \| [`{ examplename }.{ object_type_folder_name }.json`](./examples/{ examplename }.{ object_type_folder_name }.json)| )
+            ( `` )
+        ) ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT.
+
+      ELSE.
+        readme = VALUE rswsourcet(
+            ( |# { object-object_type } File Format| )
+            ( `` )
+            ( `File | Cardinality | Definition | Schema | Example` )
+            ( `:--- | :---  | :--- | :--- | :---` )
+            ( |`<name>.{ object_type_folder_name }.json` \| 1 \| [`{ interfacename }.intf.abap`](./type/{ interfacename }.intf.abap) \| [`{ to_lower( objecttype ) }-v{ format_version }.json`](./{ to_lower( objecttype ) }-v{ format_version }.json)| &&
+              | \| | )
+            ( `` )
+        ) ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT.
```

Comment:
Just a minor.
As far as I see, these lines are nearly the same. Maybe we can do it like this:
`data example_part type string.
IF object-example  IS NOT INITIAL.
        data(examplename)= replace_names_in_string( content_as_string = to_lower( object-example ) replacing_table_string = replacing_table_string ).
   example_part = | \| [`{ examplename }.{ object_type_folder_name }.json`](./examples/{ examplename }.{ object_type_folder_name }.json)| .
endif.
DATA(readme) = VALUE rswsourcet(
            ( |# { object-object_type } File Format| )
            ( `` )
            ( `File | Cardinality | Definition | Schema | Example` )
            ( `:--- | :---  | :--- | :--- | :---` )
            ( |`<name>.{ object_type_folder_name }.json` \| 1 \| [`{ interfacename }.intf.abap`](./type/{ interfacename }.intf.abap) \| [`{ to_lower( objecttype ) }-v{ format_version }.json`](./{ to_lower( objecttype ) }-v{ format_version }.json)| &&
               example_part  )
            ( `` )
        ) ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT. `

Reply:
Done